packages/gitbook/src/components/Search/useSearch.tsx

```diff
@@ -25,6 +25,21 @@ export type UpdateSearchState = (
     update: React.SetStateAction<SearchState | null>
 ) => Promise<URLSearchParams>;
 
+// Imperative setter reference to allow non-React code to update search state
+let externalSetSearchState: UpdateSearchState | null = null;
```

Comment:
Then just zustand .... we have a global state solution that e use in a bunch of places; why would we want to hardcode something like this?